tests/Unit/PointwiseFunctions/AnalyticData/CurvedWaveEquation/Test_PureSphericalHarmonic.cpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Framework/Pypp.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+void test_create_from_options() {
+  const auto option_parsed = TestHelpers::test_creation<
+      CurvedScalarWave::AnalyticData::PureSphericalHarmonic>(
+      "Radius: 0.8 \n"
+      "Width: 2. \n"
+      "l_mode: 10\n"
+      "m_mode: 8\n");
+  const CurvedScalarWave::AnalyticData::PureSphericalHarmonic constructed{
+      0.8, 2., 10, 8};
+  CHECK(option_parsed == constructed);
+  CHECK_FALSE(option_parsed != constructed);
+  test_serialization(constructed);
+}
+
+// `check_with_random_values` does not allow us to forward the l and m arguments
+// of type size_t / int to python, so we generate the random numbers and call
+// the python function manually here
+void test_variables() {
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<double> x_dist{1., 10.};
+  std::uniform_real_distribution<double> radius_dist{1., 10.};
+  std::uniform_real_distribution<double> width_dist{2., 20.};
+  // size of the DataVector we are testing with
+  const size_t dv_size = 10;
+  for (size_t l = 0; l < 8; l++) {
+    for (int m = -l; m < static_cast<int>(l); m++) {
+      const auto x = make_with_random_values<tnsr::I<DataVector, 3>>(
+          make_not_null(&gen), make_not_null(&x_dist), DataVector(dv_size));
+      const double radius = radius_dist(gen);
+      const double width = width_dist(gen);
+      const auto py_res = pypp::call<Scalar<DataVector>>(
+          "PointwiseFunctions.AnalyticData.CurvedWaveEquation."
+          "PureSphericalHarmonic",
+          "pi", x, radius, width, l, m);
+
+      const CurvedScalarWave::AnalyticData::PureSphericalHarmonic sh{
+          radius, width, l, m};
+      const auto sh_res = sh.variables(x, 0., {});
+
+      CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Pi>(sh_res), py_res);
+      CHECK(get<CurvedScalarWave::Phi<3>>(sh_res) ==
+            make_with_value<CurvedScalarWave::Phi<3>::type>(dv_size, 0.));
+      CHECK(get<CurvedScalarWave::Psi>(sh_res) ==
+            make_with_value<CurvedScalarWave::Psi::type>(dv_size, 0.));
+    }
+  }
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticData.CurvedWaveEquation.SphericalHarmonic",
+    "[Unit][PointwiseFunctions]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{""};
```

Comment:
(optional) You can configure the path here so you don't have to write it in `pypp::call`